build-system/git.js

```diff
@@ -24,55 +24,45 @@ const {getStdout} = require('./exec');
 
 const commitLogMaxCount = 100;
 
-/**
- * Returns the merge base of the current branch off of master when running on
- * a local workspace.
- * @return {string}
- */
-exports.gitMergeBaseLocalMaster = function() {
-  return getStdout('git merge-base master HEAD').trim();
-};
-
 /**
  * Returns the merge base of the current branch off of master, regardless of
  * the running environment.
  * @return {string}
  */
 exports.gitMergeBaseMaster = function() {
   if (process.env.TRAVIS) {
-    const commitRange = process.env.TRAVIS_COMMIT_RANGE.split('...');
-    return getStdout(`git merge-base ${commitRange[0]} ${commitRange[1]}`)
-        .trim();
+    return getStdout(
+        `git merge-base master ${process.env.TRAVIS_PULL_REQUEST_SHA}`).trim();
   }
-  return exports.gitMergeBaseLocalMaster();
+  return gitMergeBaseLocalMaster();
 };
 
 /**
  * Returns the `master` parent of the merge commit (current HEAD) on Travis.
  * @return {string}
  */
 exports.gitTravisMasterBaseline = function() {
-  return process.env.TRAVIS_COMMIT_RANGE.split('...')[0];
+  return getStdout('git rev-parse origin/master').trim();
```

Comment:
This line is the crux of this PR, and you've tested it across a few PR jobs. ✓

Reply:
For posterity I will add a comment with the tests below